components/omega/doc/design/VerticalMixingCoeff.md

```diff
@@ -0,0 +1,182 @@
+
+### 4.1 Data types and parameters 
+
+#### 4.1.1 Parameters
+
+The following config options should be included for the Richardson number dependent -- Pacanowski and Philander (1981) based -- vertical mixing:
+
+1. `EnableShearMix`. If true, shear-based mixing is computed based upon Pacanowski and Philander (1981) and applied to velocity components and tracer quantities. \[.true./.false.\]
+2. `ShearNuZero`. Numerator of Pacanowski and Philander (1981) Eq (1). \[0.005 $m^2 s^{-1}$\]
+3. `ShearAlpha`. Alpha value used in Pacanowski and Philander (1981) Eqs (1) and (2). \[5\]
+4. `ShearExponent`. Exponent used in denominator of Pacanowski and Philander (1981) Eqs (1). \[2\]
+
+The following config options should be included for convective vertical mixing:
+
+1. `EnableConvectiveMix`. If true, convective mixing is computed and applied to velocity components and tracer quantities. \[.true./.false.\]
+2. `ConvectiveDiffusivity`. Convective vertical viscosity and diffusivity applied to velocity components and tracer quantities, respectively. \[1.0 $m^2 s^{-1}$\]
+3. `ConvectiveTriggerBVF`. Value of Brunt Viasala frequency squared below which convective mixing is triggered. \[0.0\]
+
+The following config options should be included for background vertical mixing:
+
+1. `BackgroundViscosity`. Background vertical viscosity applied to velocity components. \[1.0e-4 $m^2 s^{-1}$\]
+2. `BackgroundDiffusivity`. Background vertical diffusivity applied to all tracer quantities. \[1.0e-5 $m^2 s^{-1}$\]
+
+### 4.2 Methods
+
+<!--- List and describe all public methods and their interfaces (actual code for
+interface that would be in header file). Describe typical use cases. -->
+
+It is assumed that viscosity and diffusivity will be stored at cell centers. Additionally, since they will be used elsewhere, it is assumed that zMid and N2 have been computed elsewhere (N2 in the density routine) and stored. Quantities such as squared shear and Richardson number are not be needed outside of the vertical mixing routines, so are computed locally.
+
+```c++
+parallelFor(
+    {NCellsAll, NVertLevels}, KOKKOS_LAMBDA(int ICell, int K) {
+
```

Comment:
How are we going to manage topography? Presumably masking to utilize the full NVertLevels.

Reply:
Currently in MPAS-O, topography for the vertical mixing is done by utilizing `minLevelCell` and `maxLevelCell`... is there an example in Omega yet for how we are doing masking? If so, I can change the example code here to reflect that.

Reply:
@katsmith133 From the meeting last week, it sounded like Brian was doing some tests to explore options. So we shouldn't let this hold up the document. Whatever we decide will end up impacting anything with a vertical index.

Reply:
Ok, sounds good. I will just make note then about adopting whatever method is decided for topography.